Shehzad account portal docs (#150)
clerkinc/clerk-docs: docs/features/account-portal.mdx

import { CodeBlockTabs } from "@/components/CodeBlockTabs";

# Introduction

> The Account Portal offers a comprehensive solution for managing user authentication and profile management in your web application and is the fastest way to add authentication to your application with minimal code required. 

Author: @BRKalow - in the context of Clerk though, it is the fastest way to add auth. The only other paths to use Clerk is to set up your sign in and sign up pages OR use our customer flows. I do like the trimming down but think that i want to reinforce that it is the fastest way to go about doing it.

> The Account Portal offers a comprehensive solution for managing user authentication and profile management in your web application and is the fastest way to add authentication to your application with minimal code required. 

Author: I'm using the commit suggestion from git so i can take your changes in, but will add the fastest way piece back from local after.

> The Account Portal offers a comprehensive solution for managing user authentication and profile management in your web application and is the fastest way to add authentication to your application with minimal code required. 

Reviewer: @msakbar Can we say something like:

> Clerk's Account Portal offers a comprehensive solution for managing user authentication and profiles in your web application with minimal code. It is the fastest way to integrate Clerk's authentication into your application.

I think it's important we scope the "fastest" claim to our product.

> The Account Portal offers a comprehensive solution for managing user authentication and profile management in your web application and is the fastest way to add authentication to your application with minimal code required. 

Author: Done - removed the double mention of Clerk.

<Image />

# Getting Started

> To use the Account Portal, simply add Clerk to your application and protect your pages. Any time an unauthenticated user attempts to access a protected page in your application, we will automatically redirect them to the Account Portal pages to Sign in.

Reviewer: nit, remove adverbs that trivialize effort, and prefer to reference Clerk in the third-person, instead of "we":

```suggestion
To use the Account Portal, add Clerk to your application and protect your pages. Any time an unauthenticated user attempts to access a protected page in your application, they will automatically get redirected to the Account Portal pages to Sign in.
```

> Depending on your choice of framework, you can redirect to them from your application using our NextJS [authMiddleware](https://clerk.com/docs/nextjs/middleware), React [Control Components](https://clerk.com/docs/component-reference/redirect-to-sign-in) or [ClerkJS SDK](https://clerk.com/docs/reference/clerkjs/clerk).

Reviewer: ```suggestion
Depending on your choice of framework, you can redirect to the Account Portal from your application using Clerk's Next.js [authMiddleware](https://clerk.com/docs/nextjs/middleware), React [Control Components](https://clerk.com/docs/component-reference/redirect-to-sign-in) or [ClerkJS SDK](https://clerk.com/docs/reference/clerkjs/clerk).
```

Note that when you go to production, this is why we require adding the `accounts` CNAME record as part of your DNS settings. If you do not want to do this and are using your own components, you may also disable the Account Portal pages via the 'Danger' tab.
</Callout>

## Custom Redirects

With the Account Portal, we default the user back to where they came from after a successful auth event (sign in or sign up) and pass this parameter via a `redirect_url` query param.

For situations where this may not be possible, we default to your application homepage and expect the appropriate redirects will kick in, but you can also specify default fallbacks directly in the dashboard as well. 

- **After sign-up fallback**: Define the landing page or a specific section of your application where users should be redirected to after successfully creating a new account.
- **After sign-in fallback**: Define the landing page or a specific section of your application where users should be directed after a successful sign-in.
- **After Logo Click**: Specify the destination page or action users should be redirected to after they click on your logo in the pages.

## Customization

> The account portal also offers a limited set of capability to align the visual appearance of the pages with your application's branding and design which includes the ability to customize the colors, font and button weights.

Reviewer: ```suggestion
The Account Portal also offers a limited set of capabilities to align the visual appearance of the pages with your application's branding and design. This includes the ability to customize the colors, font and button weights.
```
